pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -347,14 +321,20 @@ func (plugin *kubenetNetworkPlugin) SetUpPod(namespace string, name string, id k
 	shaper := plugin.shaper()
 
 	if egress != nil || ingress != nil {
-		ipAddr := plugin.podIPs[id]
-		if err := shaper.ReconcileCIDR(fmt.Sprintf("%s/32", ipAddr), egress, ingress); err != nil {
+		if err := shaper.ReconcileCIDR(fmt.Sprintf("%s/32", ip4.String()), egress, ingress); err != nil {
 			return fmt.Errorf("Failed to add pod to shaper: %v", err)
 		}
 	}
 
-	plugin.syncHostportsRules()
-	return nil
+	plugin.podIPs[id] = ip4.String()
+
+	// Open any hostports the pod's containers want
+	runningPods, err := plugin.getRunningPods()
+	if err == nil {
+		err = plugin.hostportHandler.OpenPodHostportsAndSync(pod, BridgeName, runningPods)
```

Comment:
I don't think the pod will actually get stuck, since eth0 will get destroyed along with the infra container's network namespace.  But what won't be cleaned up is any IPAM leases or shaper stuff.